server/leafnode.go

```diff
@@ -1277,6 +1307,21 @@ func (c *client) updateSmap(sub *subscription, delta int32) {
 // Send the subscription interest change to the other side.
 // Lock should be held.
 func (c *client) sendLeafNodeSubUpdate(key string, n int32) {
+	// If we are a spoke, we need to check if we are allowed to send this subscription over to the hub.
+	if c.isSpokeLeafNode() {
+		checkPerms := true
+		if len(key) > 0 && key[0] == '$' || key[0] == '_' {
```

Comment:
You forgot parenthesis here which would cause a panic should `key` be an empty string (which it won't be, but..):
```
if len(key) > 0 && (key[0] == '$' || key[0] == '_') {
..
```

Reply:
thx will fix.